app/models/concerns/fog_extensions/vsphere/server.rb

```diff
@@ -35,7 +35,14 @@ def scsi_controller_type
       end
 
       def select_nic(fog_nics, nic)
-        fog_nics.detect {|fn| fn.network == nic.compute_attributes['network']} # grab any nic on the same network
+        nic_attrs = nic.compute_attributes
+        selected_nic =   fog_nics.detect { |fn| fn.network == nic_attrs['network'] } # grab any nic on the same network
+        selected_nic ||= if service.get_network(nic_attrs['network'], datacenter).key?(:id)
+                           fog_nics.detect { |fn| fn.network  == service.get_network(nic_attrs['network'], datacenter)[:id]  } # no network? try the portgroup
+                         else
```

Comment:
rubocop (style checker)'s reporting that the "else nil" bit is superfluous, just remove these two lines